remote/remote_state_s3.go

```diff
@@ -123,6 +124,17 @@ func configValuesEqual(config map[string]interface{}, existingBackend *Terraform
 		} else {
 			terragruntOptions.Logger.Printf("Remote state configuration encrypt contains invalid value %v, should be boolean.", existingBackend.Config["encrypt"])
 		}
+
+		// If other keys are bools, DeepEqual also will consider the maps to be different.
+		for key, value := range existingBackend.Config {
+			if _, isBool := value.(bool); isBool {
+				continue // We know this is already converted to a bool, e.g. encrypt
+			}
+
+			if convertedValue, err := strconv.ParseBool(value.(string)); err == nil {
```

Comment:
Will this convert every string "true" and "false" to a boolean true and false?

Reply:
Yes.

Reply:
BTW, I corrected the implementation in the tests commit and it will now only convert to a bool if the new config is a bool.